### PR TITLE
[RFR] Lerna publication

### PR DESCRIPTION
**Pull request by alexisjanvier**
capture d ecran de 2018-07-06 09-28-27
capture d ecran de 2018-07-06 09-24-34
capture d ecran de 2018-07-06 09-24-02

### CODE REVIEW

**Comment by alexisjanvier** [comment]
For Lerna to work well with [typescript](https://github.com/Quramy/lerna-yarn-workspaces-example#resolve-dependencies-as-typescript-modules), we have to use [scoped packages](https://docs.npmjs.com/getting-started/scoped-packages). To publish on a public scoped packages, we need the [credentials](https://www.npmjs.com/docs/orgs/publishing-an-org-scoped-package.html), and I only have them on my npm username `@alexisjanvier`... I'll ask tomorrow to @fzaninotto to be added to the npm marmelab team, if it's possible.

File: packages/builder/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@alexisjanvier/postgres-queries-builder",
```
Comment (jdemangeon) [inline packages/builder/package.json]: bad name?

File: packages/builder/src/queries/insert/BatchInsert.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const batchInsert = ({
         const getValueSubQuery = valueSubQuery(writableCols);
 
         const values = rows
-            .map((row, index) => getValueSubQuery(index + 1))
+            .map((__, index) => getValueSubQuery(index + 1))
```
Comment (jdemangeon) [inline packages/builder/src/queries/insert/BatchInsert.ts]: 2 underscores ?

File: packages/builder/src/queries/select/SelectByOrderedIdentifiers.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export const selectByOrderedIdentifiers = ({
 FROM ${table}
 JOIN (
 VALUES ${values
-        .map((value, index) => `($${primaryKey}${index + 1}, ${index + 1})`)
+        .map((__, index) => `($${primaryKey}${index + 1}, ${index + 1})`)
```
Comment (jdemangeon) [inline packages/builder/src/queries/select/SelectByOrderedIdentifiers.ts]: 2 underscores ?

File: packages/pool/src/helpers.ts [hunk 4]
```diff
@@ -31,7 +31,7 @@ export const namedToNumericParameter = (
         throw new Error(`Missing Parameters: ${unmatchedTokens.join(', ')}`);
     }
 
-    const sql = fillTokens.reduce((partiallyInterpolated, token, index) => {
+    const sql = fillTokens.reduce((partiallyInterpolated, __, index) => {
```
Comment (jdemangeon) [inline packages/pool/src/helpers.ts]: 2 underscores ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 5]
```diff
@@ -106,7 +106,7 @@ export const getMatch = (filters, searchableCols, whereParts = []) => {
           );
 };
 
-export const getLike = (filters, searchableCols, whereParts = []) => {
+export const getLike = (filters, __, whereParts = []) => {
```
Comment (jdemangeon) [inline packages/builder/src/helpers/WhereQuery.ts]: 2 underscores ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 6]
```diff
@@ -126,7 +126,7 @@ export const getNotLike = (filters, searchableCols, whereParts = []) => {
     );
 };
 
-export const getFrom = (filters, searchableCols, whereParts = []) => {
+export const getFrom = (filters, __, whereParts = []) => {
```
Comment (jdemangeon) [inline packages/builder/src/helpers/WhereQuery.ts]: 2 underscores ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 7]
```diff
@@ -139,7 +139,7 @@ export const getFrom = (filters, searchableCols, whereParts = []) => {
     );
 };
 
-export const getTo = (filters, searchableCols, whereParts = []) => {
+export const getTo = (filters, __, whereParts = []) => {
```
Comment (jdemangeon) [inline packages/builder/src/helpers/WhereQuery.ts]: 2 underscores ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 8]
```diff
@@ -152,7 +152,7 @@ export const getTo = (filters, searchableCols, whereParts = []) => {
     );
 };
 
-export const getNot = (filters, searchableCols, whereParts = []) => {
+export const getNot = (filters, __, whereParts = []) => {
```
Comment (jdemangeon) [inline packages/builder/src/helpers/WhereQuery.ts]: 2 underscores ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 9]
```diff
@@ -166,7 +166,7 @@ export const getNot = (filters, searchableCols, whereParts = []) => {
     );
 };
 
-export const getQuery = (filters, searchableCols, whereParts = []) => {
+export const getQuery = (filters, __, whereParts = []) => {
```
Comment (jdemangeon) [inline packages/builder/src/helpers/WhereQuery.ts]: 2 underscores ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 10]
```diff
@@ -176,7 +176,7 @@ export const getQuery = (filters, searchableCols, whereParts = []) => {
     );
 };
 
-export const getResult = (filters, searchableCols, whereParts = []) => {
+export const getResult = (__, ___, whereParts = []) => {
```
Comment (jdemangeon) [inline packages/builder/src/helpers/WhereQuery.ts]: 2 & 3 underscores ?

File: packages/pool/package.json [hunk 11]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@alexisjanvier/postgres-queries-pool",
```
Comment (jdemangeon) [inline packages/pool/package.json]: bad name?

File: packages/postgres-queries/package.json [hunk 12]
```diff
@@ -1,4 +1,17 @@
 {
-    "name": "postgres-queries",
-    "version": "0.1.0"
+  "name": "postgres-querie",
```
Comment (jdemangeon) [inline packages/postgres-queries/package.json]: Why not postgres-queries?

File: packages/postgres-queries/package.json [hunk 13]
```diff
@@ -1,4 +1,17 @@
...
+  "main": "lib/index.js",
+  "typings": "lib/index.d.ts",
+  "scripts": {
+    "prepare": "tsc -p tsconfig.build.json"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "MIT",
+  "dependencies": {
+    "@alexisjanvier/postgres-queries-builder": "^0.1.6",
```
Comment (jdemangeon) [inline packages/postgres-queries/package.json]: bad names?

File: packages/postgres-queries/src/index.ts [hunk 14]
```diff
@@ -0,0 +1,2 @@
+export * from '@alexisjanvier/postgres-queries-builder';
```
Comment (jdemangeon) [inline packages/postgres-queries/src/index.ts]: bad names?

File: tsconfig.json [hunk 15]
```diff
@@ -1,14 +1,9 @@
...
-        "declaration": true,
-    },
-    "exclude": [
-        "node_modules",
-        "**/*.spec.ts"
-    ]
-}
+      "baseUrl": "./packages",
+      "paths": {
+        "@alexisjanvier/*": ["./*/src"]
```
Comment (jdemangeon) [inline tsconfig.json]: bad name?

**Comment by alexisjanvier** [comment]
@jdemangeon Done !

File: lerna.json [hunk 16]
```diff
@@ -5,5 +5,5 @@
   "packages": [
     "packages/*"
   ],
-  "version": "0.1.3"
```
Comment (djhi) [inline lerna.json]: I think a good practice is to add `"private": true` in this file

File: lerna.json [hunk 17]
```diff
@@ -0,0 +1,9 @@
+{
+  "lerna": "2.11.0",
```
Comment (djhi) [inline lerna.json]: Add the `private` key ?

File: packages/builder/src/helpers/WhereQuery.ts [hunk 18]
```diff
@@ -106,7 +106,7 @@ export const getMatch = (filters, searchableCols, whereParts = []) => {
           );
 };
 
-export const getLike = (filters, searchableCols, whereParts = []) => {
+export const getLike = (filters, _, whereParts = []) => {
```
Comment (djhi) [inline packages/builder/src/helpers/WhereQuery.ts]: Why this change ?

Comment (alexisjanvier) [reply packages/builder/src/helpers/WhereQuery.ts]: `searchableCols` is not used. Now `.tsconfig` says: 

```
"noFallthroughCasesInSwitch": true,
"noImplicitReturns": true,
"noUnusedLocals": true,
"noUnusedParameters": true,
```
It's not an obligation, but I thought it was in the direction of a cleaner code.

File: lerna.json (hunk 17 above)
Comment (alexisjanvier) [reply lerna.json]: Done (but I couldn't find a reference to this `private` key in the Lerna documentation)

Comment (djhi) [reply lerna.json]: Oups, sorry, I thought it was the root package.json...
